src/record.rs

```diff
@@ -197,12 +226,16 @@ impl Record {
         self.header.record_type
     }
 
+    pub fn header_type_str(&self) -> &str {
+        std::str::from_utf8(&self.header.record_type).unwrap()
```

Comment:
Given that this is basically just a utility function to allow comparisons against strings rather than slices, would it be preferable to use `String::from_utf8_lossy`? We know that any valid type will be parsed correctly, and invalid types won't match properly due to the error character that replaces the non-utf8 value. You'd still be able to access the exact value via the `header_type` function if it's important to display the exact underlying data. 

That being said, that would require unnecessary copying (to avoid the Cow enum wrapping the result, anyway), compared to simply matching against the Result of `from_utf8`.

I might try playing around with getting the Utf8Errors to nicely propagate when parsing the records, but, on the other hand, it would be a lot simpler just to return the Result.